ruby/object_oriented_programming_basics/managing_ruby_projects.md

```diff
@@ -0,0 +1,239 @@
+
+Another thing to keep in mind is that local variables do not get loaded, so if your `foo.rb` had a local variable `my_favourite_food`, trying to access it in `main.rb` would raise an error. Constants do get loaded however, so you can access those.
+
+### Gems and you
+
+Now that you know how to work with your own files it is time to learn how to work with files of others. Gems are packages that contain Ruby library or an app that someone wrote - so basically, some code. Some of those gems are part of the Ruby standard library but most of them require installing them on your own. If you use a gem then you call such a gem a dependency - your code depends on that gem to function properly. Some dependencies are only used in particular contexts, for example you can have a set of gems that are used only in a development environment or in a test environment. Many gems depend upon other gems, sometimes the versions they depend on, differ. You *could* manage gems and their dependencies on your own, going through the process of installling, updating them and resolving whatever conflicts might arise but Rubyists have tools for that!
+
+RubyGems has been part of Ruby standard library since version 1.9 and is used to get those amazing gems onto your computer. Remember that bit about `require` going through installed gems to potentially find a file you're looking for? That's the work of RubyGems. Another cool part about it? It is a gem itself! Let's give it a try.
+
+Create a new Ruby file `main.rb` in a directory called `colorful`:
+
+```rb
+require 'colorize'
+
+puts 'Red goes faster!'.colorize(:red)
+
+puts "I'm blue da ba dee da ba di!".colorize(:blue)
+
+puts "It ain't easy bein' green...".colorize(:green)
+```
+
+You're probably itching to see all those colours, so run your file with `ruby foo.rb` to see them... or rather, a LoadError. Right - you need to <span id="install-gem">install that gem first!</span> Do that with `gem install colorize` and you'll see RubyGems in action. Your system now has access to the `Colorize` gem!
+
+Wait, *your system* - what about others that would like to use your code? Yeah, they would need to `gem install` it as well, no big deal. But what if you have dozens of gems? How do you make sure that the versions you use are the versions others download? This sounds rather tedious. Enter: Bundler. Another gem, part of RubyGems but released independently. <span id="bundler">Bundler allows you to declare what gems your project needs - down to their version.</span> As for others, Bundler allows them to take that declaration, a simple file called `Gemfile`, and use it to install those gems in a quick `bundle install`. Since gem installs are global, <span id="bundle-exec">you need a way to run only those particular gem versions that are declared in the `Gemfile`.</span> You can do that by using `bundle exec` followed by a command that you want to execute - most likely `bundle exec ruby foo.rb`.
```

Comment:
Few tweaks for readability - down with big paragraphs 😆 

```suggestion
Wait, *your system* - what about others who would like to use your code? Yeah, they would also need to `gem install` it - no big deal. 

But what if you have dozens of gems? How do you ensure that the versions you use are the same version others download? This sounds rather tedious. Enter: Bundler. It's another gem, part of RubyGems, but released independently. 

<span id="bundler">Bundler allows you to declare what gems your project needs - down to their version.</span> As for others, Bundler allows them to take that declaration, a simple file called `Gemfile`, and use it to install those gems in a quick `bundle install`. 

Since gem installs are global, <span id="bundle-exec">you need a way to run only those particular gem versions that are declared in the `Gemfile`.</span> You can do that by using `bundle exec` followed by a command you want to execute - most likely `bundle exec ruby foo.rb`.
```